pkg/common/redis_test.go

```diff
@@ -87,12 +87,133 @@ func TestRedisLockWithTTLAndRetry(t *testing.T) {
 }
 
 func TestCopyStruct(t *testing.T) {
-	options1 := &types.RedisConfig{ClientName: "hello", PoolSize: 10, ConnMaxLifetime: time.Second}
-	options2 := &types.RedisConfig{}
+	tests := []struct {
+		name     string
+		src      any
+		dest     any
+		expected any
+	}{
+		{
+			name: "copy struct basic",
+			src: &types.RedisConfig{
+				ClientName:      "hello",
+				PoolSize:        10,
+				ConnMaxLifetime: time.Second,
+			},
+			dest: &types.RedisConfig{},
+			expected: &types.RedisConfig{
+				ClientName:      "hello",
+				PoolSize:        10,
+				ConnMaxLifetime: time.Second,
+			},
+		},
+		{
+			name: "copy struct overwriting fields on destination",
+			src: &types.RedisConfig{
+				ClientName:      "hello",
+				PoolSize:        10,
+				ConnMaxLifetime: time.Second,
+			},
+			dest: &types.RedisConfig{
+				ClientName: "world",
+				PoolSize:   20,
+			},
+			expected: &types.RedisConfig{
+				ClientName:      "hello",
+				PoolSize:        10,
+				ConnMaxLifetime: time.Second,
+			},
+		},
+		{
+			name: "copy struct overwriting fields on destination part 2",
+			src: &types.Worker{
+				Id:       "123",
+				Status:   types.WorkerStatusAvailable,
+				TotalCpu: 10,
+			},
+			dest: &types.Worker{
```

Comment:
@nickpetrovic can you add a test for machine state as well

Reply:
Added.